File: samples/retrofit/app/src/main/java/com/chizi/retrofit/MainActivity.java
package com.chizi.retrofit;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import java.io.IOException;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private Button mTestButton;
    private TextView mContentTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();
        initListeners();
    }

    private void initViews() {
        mTestButton = findViewById(R.id.btnTest);
        mContentTextView = findViewById(R.id.tvContent);
    }

    private void initListeners() {
        mTestButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getRepos();
            }
        });
    }

    private void getRepos() {
        new Thread(new Runnable() {
            @Override
            public void run() {
                try {
                    List<RepoBean> repoList = GithubService.newInstance().getRepos("chizidegit");
                    setResultData(repoList);
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }).start();
    }

    private void setResultData(final List<RepoBean> result) {
        runOnUiThread(new Runnable() {
            @Override
            public void run() {
                mContentTextView.setText(result.toString());
            }
        });
    }
}

File: samples/constraintlayout/app/src/main/java/com/bytedance/chizi/constraint/layout/MainActivity.kt
package com.bytedance.chizi.constraint.layout

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), View.OnClickListener {
    override fun onClick(v: View) {
        when (v.id) {
            R.id.tv_action -> {
                val intent = Intent(this, Main2Activity::class.java)
                startActivity(intent)
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tv_action.setOnClickListener(this)
    }
}
